lib/rails-jquery-autocomplete/autocomplete.rb

```diff
@@ -97,13 +97,19 @@ def get_object(model_sym)
     # Hash also includes a key/value pair for each method in extra_data
     #
     def json_for_autocomplete(items, method, extra_data=[])
-      items = items.collect do |item|
-        hash = HashWithIndifferentAccess.new({"id" => item.id.to_s, "label" => item.send(method), "value" => item.send(method)})
-        extra_data.each do |datum|
-          hash[datum] = item.send(datum)
-        end if extra_data
-        # TODO: Come back to remove this if clause when test suite is better
-        hash
+      if items.first && items.first.kind_of?(Hash)
+        items.each do |item|
+          item['label'] = item['value'] = item[method.to_s]
+        end
+      else
+        items = items.collect do |item|
+          hash = HashWithIndifferentAccess.new({"id" => item.id.to_s, "label" => item.send(method), "value" => item.send(method)})
```

Comment:
Line is too long. [130/80]<br>Redundant curly braces around a hash parameter.<br>Space inside { missing.<br>Space inside } missing.